src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.DateTime.netfx.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+
+namespace System.Diagnostics
+{
+    partial class Activity
+    {
+        private DateTime GetUtcNow()
+        {
+            // Timer ticks need to be converted to DateTime ticks
+            long dateTimeTicksDiff = (long)((Stopwatch.GetTimestamp() - syncStopwatchTicks) * 10000000L /
+                                   (double)Stopwatch.Frequency);
+            
+            // DateTime.AddSeconds (or Milliseconds) rounds value to 1 ms, use AddTicks to prevent it
+            return syncUtcNow.AddTicks(dateTimeTicksDiff);
+        }
+
+        private static void Sync()
+        {
+            // wait for DateTime.UtcNow update to the next granular value
+            Thread.Sleep(1);
+            syncStopwatchTicks = Stopwatch.GetTimestamp();
+            syncUtcNow = DateTime.UtcNow;
+        }
+
+        private static DateTime syncUtcNow;
```

Comment:
We have a ugly problem the first time GetUtcNow is run.   I checked, and the Timer always spawns the callback on another thread (even in the 0 delay case).  This means that the construction of 'syncTimeUpdater' does not insure that syncUtcNow and syncStopWatch ticks are initialized (the constructor returns before the callback is raised).  Because Sync() has a sleep, this means the first use of GetUtcNow is almost certainly to use uninitialized values.    I ran an experiment and indeed the first time I get back from GetUtcNow() is close to DateTime.MinValue, which is 'wildly' off.  

Note that we also don't want to force the first GetUtcNow to be synchronous with the call to Sync() since that would stall that call by up to 16 msec the first time it is called, and we don't want measurement to cause non-trivial delays. 

My recommendation is to simply initialize syncUtcNow to  add initialization with this code.  

    // Initialize them for the first call.   It will not be well synced (as Sync() does), but it is the best we can 
    // do without blocking for up to 16 msec like Sync() does.    These values will only be used in the first
    // 16 msec after the first call to GetUtcNow() because Sync will update them at that time.  
private static DateTime syncUtcNow = DateTime.UtcNow; 
private static long syncStopwatchTicks = Stopwatch.GetTimestamp();

The good news here is that when Sync() ultimately does the synchronization, the high likelihood is that the better value will make time 'jump forward' and not backward.   This is good because it avoids negative time deltas, which often will confuse upper level logic.